### ISSUE TITLE
connect to intake catalog

### ISSUE THREAD

**Issue by ibnbay99** [nils-braun/dask-sql#99]
- Can dask-sql connect to [intake](https://github.com/intake/intake) catalog server and using their own `Authorization Plugins`?

I would love if this 2 project can merge their great features, to be more powerful tools 🙂

**Comment by ibnbay99** [comment]
May be it will be great connect to apache atlas for more connection to others app

**Comment by nils-braun** [comment]
That is a very nice feature request! 
I think that should already work "out of the box" if you are using python syntax (as intake is able to export dask dataframes, which then can be used in dask-sql after registration as table).
It would be super cool if we can even do this from SQL, e.g.
```
CREATE TABLE "test" WITH (format = "intake", location = "intake-catalog-location")
```
or something like this. What do you think?

For Apache Atlas I have not much experience - do you know if it is possible to connect from dask to Apache Atlas?

**Comment by ibnbay99** [comment]
Apache Atlas can connect to python, and `may be` reuse code connect to Hive currently used.

**Comment by nils-braun** [comment]
That would be very great! Could you share a typical workflow with Apache Atlas and the python client? As I said, I do not have much experience with Apache Atlas (sorry..) and for me Atlas is mostly a tool for Audit-safety and compliance (not so much as a metadata catalog by itself).

**Comment by martindurant** [comment]
```
CREATE TABLE "test" WITH (format = "intake", location = "intake-catalog-location")
```

what do you need from Intake to make this happen? I assume we require the named data source to be of "dataframe" type, and will use it by calling `to_dask()`.

**Comment by nils-braun** [comment]
Without having workes with intake before (but fortunately the docs are good ;-)), I assume all dask-sql would need to do on a call like
```sql
CREATE TABLE "test" WITH (format = "intake", location = "intake-catalog-location", table_name = "my-table")
-- table_name could be non-mandatory and would use "test" otherwise
```
would be 
```python
cat = intake.open_catalog("intake-catalog-location")
df = cat["my-table"].to_dask()
```
Is this correct? If this is the case, that would already be everything :-)

**Comment by martindurant** [comment]
That is correct - although it's possible to optionally allow extra arguments when opening the catalog or accessing the specific dataset.

**Comment by nils-braun** [comment]
@ibnbay99 - the intake part is included now. Do you have a code sample for Apache Atlas?
